### ISSUE TITLE
Improved Sprite editor

### ISSUE THREAD

**Issue by Hawf** [YoYoGames/GameMaker-Feature-Requests#1527]
### Is your feature request related to a problem?

One of the nicest features of GMS1's sprite editor is how it had a lot of simple manipulation tools for easy editing.
The GMS2 sprite editor feels like an enormous step back and was a huge disappointment the first time I used it.
The bare minimum of a copy paste function is implemented in one of the most head-scratchingly obtuse ways imaginable.

Exporting sprite sheets from your paint software simply to import back into GMS2 is tedious and time consuming, especially compared to a simple ctrl+v of the individual frames that GMS1 supported.

### Describe the solution you'd like

The GMS2 sprite editor should at the very minimum have the following
- Properly implemented Ctrl+C Ctrl+V functionality, it should paste the whole sprite into the image canvas, not make a new brush.
- Horizontal and Vertical flipping of the image canvas.
- Basic Brightness, Saturation, and Hue shifting.

Ideally copy the old sprite editor wholecloth back into GMS2.

### Additional context

I implore the GMS2 dev team consider this a priority if they haven't already, I was excited to try out GMS2 when the new free version came out, but was significantly taken aback by how primitive the sprite editor was compared to GMS1, so much so that it was almost a deal breaker for me. Now while I understand many devs don't even use the GM sprite editor for their graphics, opting to use something like Graphics Gale or Aseprite, the ability to effortlessly copy paste assets and easily orient them shouldn't be undersold as a feature of GMS.

**Comment by mistletoe** [comment]
I'd rather that YYG went the other way, and _deprecated the feature entirely_ in favor of "open Sprite in Image Editor" as a right-click action. 

Other than editing the Frames in the sequencer, none of that is ever getting used over here. Photoshop, Aseprite, etc. are massively better. Turning GMS into some sort of half-baked image editor was never in anybody's interests, and I think it's a waste of resources to assign anybody to improve it when there are so many better options.

**Comment by Hawf** [comment]
> I'd rather that YYG went the other way, and _deprecated the feature entirely_ in favor of "open Sprite in Image Editor" as a right-click action.
> 
> Other than editing the Frames in the sequencer, none of that is ever getting used over here. Photoshop, Aseprite, etc. are massively better. Turning GMS into some sort of half-baked image editor was never in anybody's interests, and I think it's a waste of resources to assign anybody to improve it when there are so many better options.

I disagree, I think this is a very shortsighted and dismissive approach to this issue. having a functional asset editor is incredibly helpful, having it available for quick sprite edits in GMS1 was a godsend, removing the image editor altogether would create far more problems.

For instance, what about the developers who make ALL of their 2d game assets inside the editor inside GMS2?
There is far more to be gained by simply improving the editor we have, than to remove it completely.

**Comment by mistletoe** [comment]
I'm sorry if that came across as too harsh!

I've just re-tested the current Image Editor to see what you're saying.

1. I could copy and paste Sprites easily, via CTRL-A, CTRL-C, CTRL-V.
2. I could make a new Frame easily, via the plus sign.
3. Once copied, the data is now a Brush, which can be scaled, rotated, even used to erase with. This part is a bit confusing; when something's copied, it automatically is a Brush and it's automatically selected. If the end-user wants to go back to single-pixel editing, they can select the Brush in the Brush menu at the top right of the Image Editor UI.
4. The rotation function automatically interpolates the pixels. Being able to turn that off for pixel-art looks would be handy- there may be a way to do that, but I didn't see it.
5. Layers allow multiple images to be composited. The only (major) thing missing there is any way to filter layers; no additive or multiplicative blending.

Overall... IDK. It's about as good as I'd expect, for a super-basic editor. What feature, exactly, are you missing? Can you give a bit more description for how "effortlessly copy paste assets and easily orient them" might work better? Right now, it appears that you can copy, paste and move / orient them; my only major caveat is that rotating them appears to trigger interpolation rather than nearest-neighbor (and that's probably something we can turn off).
